Frontend/ui-driver/index.js

```diff
@@ -211,13 +219,19 @@ window.onMerchantEvent = function (_event, payload) {
       }else if (parsedPayload.payload.notificationData && parsedPayload.payload.notificationData.notification_type == "PAYMENT_MODE_MANUAL") {
         purescript.main(makeEvent("PAYMENT_MODE_MANUAL", ""))();
       } else if (parsedPayload.payload.viewParam){
-        purescript.onNewIntent(makeEvent("DEEP_VIEW", parsedPayload.payload.viewParam))();
+        if (!checkForReferral(parsedPayload.payload.viewParam, "REFERRAL_NEW_INTENT")) {
```

Comment:
Can we write it in a different function, this `onMerchantEvent` function looks very messy ?